src/emit/binary.rs

```diff
@@ -1074,6 +1077,38 @@ impl<'a> Binary<'a> {
                 .const_int(error_with_loc.len() as u64, false),
         );
     }
+
+    /// Emit encoded error data of "Panic(uint256)" as interned global string.
+    ///
+    /// On Solana, because reverts do not return data, a nil ptr is returned.
+    pub(super) fn panic_data_const(
+        &self,
+        ns: &Namespace,
+        code: PanicCode,
+    ) -> (PointerValue<'a>, IntValue<'a>) {
+        if ns.target == Target::Solana {
+            return (
+                self.context
+                    .i8_type()
+                    .ptr_type(AddressSpace::default())
+                    .const_null(),
+                self.context.i32_type().const_zero(),
+            );
+        }
+
+        let expr = Expression::NumberLiteral {
+            loc: pt::Loc::Codegen,
+            ty: Type::Uint(256),
+            value: (code as u8).into(),
+        };
+        let bytes = create_encoder(ns, false)
+            .const_encode(&[SolidityError::Panic(code).selector_expression(), expr])
+            .unwrap();
```

Comment:
You can just do `code.abi_encode` here, can't you?
Perhaps this would be cleaner.

Reply:
I see what you mean, but `SolidityError::AbiEncode` encodes any error data and not just const ones, which is why it takes a CFG. It shouldn't be used in emit, I made it `pub(super)`.